pkg/objstore/objstore.proto

```diff
@@ -0,0 +1,52 @@
+syntax = "proto3";
+
+package objstore;
+
+/* Design questions
+  - Do we want to be able to upload only part of an object
+  - What are we doing for permissions / ACLs
+  - What do we want to do for location, e.g., regions?
+  - How do object versions work?
+  - What concurrency models / consistency guarantees do we support
+  - How are we encoding error codes? Can we use standard rRPC status (https://github.com/grpc/grpc/blob/master/doc/statuscodes.md)
```

Comment:
Are there standard ways to translate those codes into client languages? If not, you can just use protobuf enums, and those are compiled into enums in the corresponding language, which helps with readability quite a bit (e.g., [here](https://github.com/hydro-project/droplet/blob/master/droplet/server/executor/call.py#L65)).

Reply:
The main advantage that I see in using the standard error codes, at least in Go, is that you check the error once.

Here's what it could end up looking like with two levels of errors:
```golang
func doPut(objectName string, objectData []byte) {
    // ... get client and context
    r, err := client.Put(ctx, &pb.PutRequest{ObjectName:objectName,ObjectData:data})
    if err != nil {
        return err
    }
    if r.ErrCode != pb.OK {
        return NewObjStoreErr(r.ErrCode)
    }
    return nil
}
```

vs

```golang
func doPut(objectName string, objectData []byte) {
    // ... get client and context
    r, err := client.Put(ctx, &pb.PutRequest{ObjectName:objectName,ObjectData:data})
    if err != nil {
        return err
    }
    return nil
}
```

Reply:
+1 for standard error codes.

Reply:
We are planning to go down the path of standard error codes. It definitely makes the code nicer. One disadvantage we should be mindful of is that the errors become a more explicit part of the interface when we define them as part of the protobuf messages. Fully specifying the behavior requires tests either way, though.